libs/gltfio/include/gltfio/MaterialProvider.h

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (C) 2019 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef GLTFIO_MATERIALPROVIDER_H
+#define GLTFIO_MATERIALPROVIDER_H
+
+#include <filament/Engine.h>
+#include <filament/Material.h>
+#include <filament/MaterialInstance.h>
+
+#include <array>
+
+namespace gltfio {
+
+enum class AlphaMode : uint8_t {
+    OPAQUE,
+    MASK,
+    BLEND
+};
+
+// NOTE: This key is processed by MurmurHashFn so please make padding explicit.
+struct alignas(8) MaterialKey {
```

Comment:
You should assert on a sizeof(MaterialKey), just to check that compilers don't add extra padding. We had to use a #pragma in other places before.

Reply:
Good idea, added a static_assert.